apix-integrationtests/src/main/java/eu/xenit/apix/util/SolrTestHelperBaseImpl.java

```diff
@@ -0,0 +1,116 @@
+package eu.xenit.apix.util;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import javax.sql.DataSource;
+import org.alfresco.repo.management.subsystems.SwitchableApplicationContextFactory;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class should only be used by the integration tests.
+ */
+public abstract class SolrTestHelperBaseImpl implements SolrTestHelper {
+    private Logger logger = LoggerFactory.getLogger(SolrTestHelperBaseImpl.class);
+
+    private DataSource dataSource;
+    protected SolrAdminClient solrAdminClient;
+
+    private int maxTries = 20;
+
+    public SolrTestHelperBaseImpl(String baseUrl, DataSource dataSource, SwitchableApplicationContextFactory searchSubSystem) {
+        this.dataSource = dataSource;
+        solrAdminClient = new SolrAdminClient(baseUrl, searchSubSystem);
+    }
+
+    /**
+     * Beware of calling this method too frequently since it runs an aggregate query over the entire alf_nodes table
+     *
+     * @return true if the latest Alfresco transaction has been indexed by Solr
+     */
+    @Override
+    public boolean areTxnsSynced() {
```

Comment:
(non-blocking) I vastly prefer `Transaction` over `Tx` in all methods and variables.